### PR TITLE
MON-3548 refacto(front): Rework dynamic components + build process

### PR DESCRIPTION
**Pull request by kduret**
* use custom webpack configuration
* use sass-loader of webpack instead of compile sass manually
* create webservice to get list of hooks by modules
* implement dynamic import of external components using systemjs

Ref: MON-3548

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "@babel/plugin-transform-arrow-functions",
+    "@babel/plugin-transform-destructuring",
+    "@babel/plugin-transform-function-name",
+    "@babel/plugin-transform-parameters",
+    "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-transform-classes",
+    "@babel/plugin-transform-shorthand-properties",
+    "@babel/plugin-transform-regenerator"
+  ]
+}
```
Comment (sc979) [inline .babelrc]: shouldn't you add a line at the end of the file ?

File: src/Centreon/Application/Webservice/CentreonFrontendComponent.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
+<?php
+/*
+ * Copyright 2005-2019 Centreon
+ * Centreon is developped by : Julien Mathis and Romain Le Merlus under
```
Comment (sc979) [inline src/Centreon/Application/Webservice/CentreonFrontendComponent.php]:
```suggestion
 * Centreon is developed by : Julien Mathis and Romain Le Merlus under
```

File: src/Centreon/Domain/Service/FrontendComponentService.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
+<?php
```
Comment (sc979) [inline src/Centreon/Domain/Service/FrontendComponentService.php]: no license ?

File: src/Centreon/Domain/Service/FrontendComponentService.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        if (is_dir($dir)) {
+            $files = scandir($dir);
+        }
+
+        foreach ($files as $key => $value) {
+            $path = $dir . DIRECTORY_SEPARATOR . $value;
+            if (!is_dir($path) && preg_match($regex, $path)) {
+                // group files by directory
+                $results[dirname($path)][] = basename($path);
+            } else if ($value != "." && $value != "..") {
```
Comment (sc979) [inline src/Centreon/Domain/Service/FrontendComponentService.php]: else if or elseif ?

File: src/Centreon/Domain/Service/FrontendComponentService.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        return $this->di;
+    }
+
+    /**
+     * Get directory files grouped by directory matching regex
+     *
+     * @param string $dir the directory to explore
+     * @param array $results the found files
+     * @param string $regex the regex to match
+     * @return void
```
Comment (sc979) [inline src/Centreon/Domain/Service/FrontendComponentService.php]: return array

File: src/Centreon/Domain/Service/FrontendComponentService.php [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * Get status for centreon instance (is master or is not master)
+     *
+     * @return array The list of pages (routes, js and css)
+     */
+    public function getPages(): array
+    {
+        return [];
```
Comment (sc979) [inline src/Centreon/Domain/Service/FrontendComponentService.php]: WIP ?

Comment (kduret) [reply src/Centreon/Domain/Service/FrontendComponentService.php]: yes

File: .babelrc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      "@babel/preset-react"
+    ],
+    [
+      "@babel/preset-env",
+      {
+        "targets": {
+          "esmodules": false
+        }
+      }
+    ],
```
[the remark this answers is not in the document]
Comment (kduret) [reply .babelrc]: yep, it allows better git diff if something is added later

**Pull request merged**